### PR TITLE
Release 3.0.0

### PR DESCRIPTION
**Pull request by rlepinski**
Updates Gimbal and Airship SDK to latest.

Testing:
- Changes have been in Goat for the last few months.

### CODE REVIEW

File: gimbal-adapter/src/main/java/com/urbanairship/gimbal/GimbalAdapter.java [hunk 1]
```diff
@@ -325,12 +334,14 @@ void updateDeviceAttributes() {
                 deviceAttributes.putAll(DeviceAttributesManager.getInstance().getDeviceAttributes());
             }
 
-            if (UAirship.shared().getNamedUser().getId() != null) {
-                deviceAttributes.put(GIMBAL_UA_NAMED_USER_ID, UAirship.shared().getNamedUser().getId());
+            String namedUserId = UAirship.shared().getNamedUser().getId();
+            if (namedUserId != null) {
+                deviceAttributes.put(GIMBAL_UA_NAMED_USER_ID, namedUserId);
```
Comment (BrianBatchelder) [inline gimbal-adapter/src/main/java/com/urbanairship/gimbal/GimbalAdapter.java:339]: Does this mean Gimbal's namedUser can never be cleared?

File: gimbal-adapter/src/main/java/com/urbanairship/gimbal/AirshipReadyReceiver.java [hunk 2]
```diff
@@ -7,14 +7,34 @@
...
         GimbalAdapter.shared(context).restore();
+
+        UAirship.shared().getPushManager().addRegistrationListener(new RegistrationListener() {
+            @Override
+            public void onChannelCreated(@NonNull String channelId) {
+                GimbalAdapter.shared(context).updateDeviceAttributes();
+            }
+
+            @Override
+            public void onChannelUpdated(@NonNull String channelId) {
```
Comment (BrianBatchelder) [inline gimbal-adapter/src/main/java/com/urbanairship/gimbal/AirshipReadyReceiver.java:31]: Maybe call `GimbalAdapter.shared(context).updateDeviceAttributes()` here also?

**Review by BrianBatchelder — COMMENTED** [review]
1 suggestion, 1 question. Minor.

File: gimbal-adapter/src/main/java/com/urbanairship/gimbal/AirshipReadyReceiver.java (hunk 2 above)
Comment (rlepinski) [reply gimbal-adapter/src/main/java/com/urbanairship/gimbal/AirshipReadyReceiver.java:31]: We only need to do it on channel create

**Review by BrianBatchelder — APPROVED** [review]
👍

**Review by crow — APPROVED** [review]
🍔

**Pull request merged**